src/components/Header/index.tsx

```diff
@@ -38,93 +33,70 @@ const Header = ({
   currentTab,
   onChangeTab,
   onToggleMenu,
-  tabs
+  tabs,
+  captureRef
 }: HeaderProps) => {
-  const [{ term, oscar, pinnedCrns }, { setTerm }] = useContext(TermContext);
+  const [
+    { term, versionName, oscar, pinnedCrns },
+    { setTerm, setVersionName }
+  ] = useContext(ScheduleContext);
   const [terms] = useContext(TermsContext);
-  const [theme, setTheme] = useContext(ThemeContext);
-  const captureRef = useRef<HTMLDivElement>(null);
+  const [versionLists, { patchVersionsData }] = useContext(VersionsContext);
+  const versionList = versionLists[term];
+  const [theme] = useContext(ThemeContext);
+  const [versionIndex, setVersionIndex] = useState(
+    versionList.indexOf(versionName)
+  );
+  const possibleVersions = [
+    'Primary',
+    'Secondary',
+    'Tertiary',
+    'Quaternary',
+    'Quinary',
+    'Senary',
+    'Septenary',
+    'Octonary',
+    'Nonary',
+    'Denary'
+  ];
+
+  const addVersion = () => {
```

Comment:
Some comments explaining the algorithm being used here would be useful

Reply:
I added comments to the function. Thanks for pointing out it was hard to understand.